src/components/PromptDialog.js

```diff
@@ -18,8 +20,14 @@ function PromptDialog(props) {
   }
 
   function handleSubmit() {
-    onClose(value);
-    setValue("");
+    if (filter.isProfane(value)) {
+      alert(
+        "We detected that your name contains profane language. If you think this was a mistake, please let us know!"
```

Comment:
Maybe this should say "input" rather than "name" since this is supposed to be a general PromptDialog?